### PR TITLE
Update Equinix configuration file

### PR DESCRIPTION
**Pull request by chrisho**
**Problem:**
Current Equinix guidelines for building a Harvester cluster fail.

**Solution:**
Optimized to update Equinix configuration files.

**Related Issue:**
harvester/harvester#2226
harvester/harvester#1181

**Test plan:**
Follow the Readme.md documentation to verify that it can be created successfully.

### CODE REVIEW

File: equinix/README.md [hunk 1]
```diff
@@ -26,6 +29,8 @@ The `harvester.install.automatic=true` parameter tells the installer we want to
 
 The `harvester.install.config_url=https://metadata.platformequinix.com/userdata` parameter tells the installer we want to fetch the Harvester configuration from this URL, which contains the userdata specified by the user when creating nodes. The harvester configuration contains sensitive credentials. We can prevent those credentials from leaking by using [userdata](https://metal.equinix.com/developers/docs/servers/user-data/) that can only be seen by provisioning nodes.
 
+**Note:** Currently ipxe on Equinix is not supported `https://release.rancher.com`, you can build your own web server or use other methods to download the relevant dependencies.
```
Comment (guangbochen) [inline equinix/README.md:32]:
```suggestion
**Note:** Due to a known [certificate issue](https://github.com/harvester/harvester/issues/2226), ipxe on Equinix currently does not support `https://release.rancher.com`, you can build your own web server or use other methods to download related artifacts.
```

File: equinix/README.md [hunk 2]
```diff
@@ -12,9 +12,12 @@ Edit the `ipxe-install` file:
 
 ```
 #!ipxe
-
-kernel https://releases.rancher.com/harvester/master/harvester-master-vmlinuz-amd64 ip=dhcp net.ifnames=1 rd.cos.disable rd.noverifyssl root=live:https://releases.rancher.com/harvester/master/harvester-master-rootfs-amd64.squashfs console=ttyS1,115200 harvester.install.automatic=true harvester.install.config_url=https://metadata.platformequinix.com/userdata
-initrd https://releases.rancher.com/harvester/master/harvester-master-initrd-amd64
+dhcp
+iflinkwait -t 5000 && echo Detected link on ${ifname} 
+set version master # change to a specific version for production installation
+set base https://releases.rancher.com/harvester/${version}  # currently ipxe on Equinix is not supported https://release.rancher.com, you can build your own web server or use other methods to download the relevant dependencies.
```
Comment (guangbochen) [inline equinix/README.md:18]:
```suggestion
set base https://releases.rancher.com/harvester/${version}  # ipxe on Equinix currently does not support `https://release.rancher.com`, you can build your own web server or use other methods to download related artifacts.
```

File: equinix/ipxe-install [hunk 3]
```diff
@@ -1,5 +1,8 @@
 #!ipxe
-
-kernel https://releases.rancher.com/harvester/master/harvester-master-vmlinuz-amd64 ip=dhcp net.ifnames=1 rd.cos.disable rd.noverifyssl root=live:https://releases.rancher.com/harvester/master/harvester-rootfs-amd64.squashfs console=ttyS1,115200 harvester.install.automatic=true harvester.install.config_url=https://metadata.platformequinix.com/userdata
-initrd https://releases.rancher.com/harvester/master/harvester-master-initrd-amd64
-boot
+dhcp
+iflinkwait -t 5000 && echo Detected link on ${ifname} 
+set version master # change to a specific version for production installation
+set base https://releases.rancher.com/harvester/${version}  # currently ipxe on Equinix is not supported https://release.rancher.com, you can build your own web server or use other methods to download the relevant dependencies.
```
Comment (guangbochen) [inline equinix/ipxe-install:5]: ditto

File: equinix/README.md [hunk 4]
```diff
@@ -40,29 +45,29 @@ The `harvester.install.config_url=https://metadata.platformequinix.com/userdata`
...
-      networks:
-        harvester-mgmt: # The management bond name. This is mandatory.
-          interfaces:
-          - name: eth0
-          default_route: true
-          method: dhcp
       device: /dev/sda
       iso_url: https://releases.rancher.com/harvester/master/harvester-master-amd64.iso
       tty: ttyS1,115200n8
+      vip: <server_public_ip> # Randomly fill in the IP first or reinstall when the Public IP is known
```
Comment (guangbochen) [inline equinix/README.md:61]: @ibrokethecloud is this the best method we can do for now of the Equinix platform installation?

File: equinix/ipxe-install (hunk 3 above)
Comment (chrisho) [reply equinix/ipxe-install]: updated

File: equinix/README.md (hunk 4 above)
Comment (ibrokethecloud) [reply equinix/README.md:61]: @guangbochen we can create an elastic IP in advance and attach it to the instance, and then use the same in vip.

This is what my sample terraform module does.

File: equinix/README.md [hunk 5]
```diff
@@ -40,29 +45,30 @@ The `harvester.install.config_url=https://metadata.platformequinix.com/userdata`
...
-          method: dhcp
       device: /dev/sda
       iso_url: https://releases.rancher.com/harvester/master/harvester-master-amd64.iso
       tty: ttyS1,115200n8
+      vip: <server_public_ip> # use Elastic IPs or reinstall when the Public IP is known
+      vip_mode: static
     ```
     - The [userdata file](./userdata-create.yaml) is included in the source and the user can modify it according to any need.
     - The Harvester installer doesn't require the `#cloud-config` line at the beginning, but Equinix Metal validates if the userdata contains it.
+    - The `vip` parameter is required to be filled.
```
Comment (guangbochen) [inline equinix/README.md:66]:
```suggestion
    - The `vip` parameter is required.
```

File: equinix/README.md [hunk 6]
```diff
@@ -40,29 +45,30 @@ The `harvester.install.config_url=https://metadata.platformequinix.com/userdata`
...
-      networks:
-        harvester-mgmt: # The management bond name. This is mandatory.
-          interfaces:
-          - name: eth0
-          default_route: true
-          method: dhcp
       device: /dev/sda
       iso_url: https://releases.rancher.com/harvester/master/harvester-master-amd64.iso
       tty: ttyS1,115200n8
+      vip: <server_public_ip> # use Elastic IPs or reinstall when the Public IP is known
```
Comment (guangbochen) [inline equinix/README.md:61]:
```suggestion
      vip: <server_public_ip> # use Elastic IP or reinstall when the public IP is known
```

File: equinix/README.md [hunk 7]
```diff
@@ -98,3 +104,12 @@ install:
...
 **NOTE:** the user can create multiple servers to join a cluster with the same userdata at once.
+
+
+### Request Elastic IPs (optional)
+- Select `IPs & Networks` from the menu bar, and then click `IP`.
+- Click `Request IP Addresses`.
+- Select `Public IPv4` item for deployment type.
+- Select the `Location` to be new.
+- The quantity must be selected to be **less than or equal to 31**.
+- Click `Submit Request` and wait until the new IP.
```
Comment (guangbochen) [inline equinix/README.md:115]:
```suggestion
### Configure Elastic IP
- Select `IPs & Networks` from the menu bar, and then click `IP`.
- Click `Request IP Addresses`.
- Select `Public IPv4` item for deployment type.
- Select the `Location` to be new.
- The quantity should be **less than or equal to 31**.
- Click `Submit Request` and wait to get the elastic IP.
```

File: equinix/README.md [hunk 8]
```diff
@@ -40,29 +45,30 @@ The `harvester.install.config_url=https://metadata.platformequinix.com/userdata`
...
       device: /dev/sda
       iso_url: https://releases.rancher.com/harvester/master/harvester-master-amd64.iso
       tty: ttyS1,115200n8
+      vip: <server_public_ip> # use Elastic IPs or reinstall when the Public IP is known
+      vip_mode: static
     ```
     - The [userdata file](./userdata-create.yaml) is included in the source and the user can modify it according to any need.
     - The Harvester installer doesn't require the `#cloud-config` line at the beginning, but Equinix Metal validates if the userdata contains it.
+    - The `vip` parameter is required to be filled.
+  - To configure Elastic IPs for Public IPv4, enable `Configure IPs`, select `Deploy from your subnet` and select that the Subnet Size must be less than or equal to 31 (optional).
```
Comment (guangbochen) [inline equinix/README.md:67]:
```suggestion
  - (optional)[configure Elastic IP](#configure-elastic-ip) and use it as the `vip`.
```

**Review by guangbochen — APPROVED** [review]
LGTM, can u squash the commit, thanks.

**Comment by chrisho** [comment]
> LGTM, can u squash the commit, thanks.

updated

**Review by ibrokethecloud — APPROVED** [review]

**Pull request merged**